File: ueueo-framework/ueueo-eventbus/src/main/java/com/ueueo/eventbus/distributed/InboxConfigDictionary.cs
using System;
using System.Collections.Generic;

namespace Volo.Abp.EventBus.Distributed;

public class InboxConfigDictionary : Dictionary<String, InboxConfig>
{
    public void Configure(Action<InboxConfig> configAction)
    {
        Configure("Default", configAction);
    }

    public void Configure(String outboxName, Action<InboxConfig> configAction)
    {
        var outboxConfig = this.GetOrAdd(outboxName, () => new InboxConfig(outboxName));
        configAction(outboxConfig);
    }
}

File: ueueo-framework/ueueo-eventbus/src/main/java/com/ueueo/eventbus/distributed/InboxConfigDictionary.java
package com.ueueo.eventbus.distributed;

import java.util.HashMap;
import java.util.function.Consumer;

public class InboxConfigDictionary extends HashMap<String, InboxConfig> {
    public void configure(Consumer<InboxConfig> configAction) {
        configure("Default", configAction);
    }

    public void configure(String outboxName, Consumer<InboxConfig> configAction) {
        InboxConfig outboxConfig = computeIfAbsent(outboxName, key -> new InboxConfig(outboxName));
        configAction.accept(outboxConfig);
    }
}
